controllers/books.controller.js

```diff
@@ -1,50 +1,70 @@
 import {Book} from '../models/book.model.js';
-import logger from '../config/logger/logger-service.js';
+
+const NO_BOOKS_FOUND_FOR_ID = 'No books found for given ID';
 
 export const BooksController = {
 
-    async getAllBooks(req, res) {
-        try {
+	async getAllBooks(req, res, next) {
+		try {
 			const result = await Book.find().sort('name');
 			res.status(200).send(result);
-		} catch (error) {
-			res.status(404).send(error);
-			logger.warn('GetAllBooks 404', req);
+		} catch (err) {
+			next(err);
 		}
-
 	},
-	
-	async getBook(req, res) {
+
+	async getBook(req, res, next) {
 		try {
-			const result = await Book.findById(req.params.id);
-			res.status(200).send(result);
-		} catch (error) {
-			res.status(404).send(error);
-			logger.warn('GetBook 404', req);
+			const result = await Book.findOne({
+				_id: req.params.id
+			});
+			if(!result) {
+				res.status(404).send(NO_BOOKS_FOUND_FOR_ID);
+			} else {
+				res.status(200).send(result);
+			}
+		} catch (err) {
+			next(err);
 		}
 	},
 
-    async updateBook(req, res) {
-        try {
-			const book = await Book.findByIdAndUpdate(req.params.id, req.body, {
-				isTakenByUser: req.params.userId
+	async updateBook(req, res, next) {
+		try {
+
+		const book = await Book.findByIdAndUpdate(
+			req.params.id, {
+				isTakenByUser: req.query.userId
+			}, {
+				new: true,
+				returnNewDocument: true,
+			},
+			function (err, result) {
+				if (err) {
+					next(err);
+				} else if (result) {
+					res.status(200).send(result);
```

Comment:
use in future res.json(). It sets correct content-type for your response.

Reply:
replaced for json